### ISSUE TITLE
Allow specifying empty docker directory for flexible staging

### ISSUE THREAD

**Issue by meltsufin** [GoogleCloudPlatform/appengine-plugins-core#275]
`CloudSdkAppEngineFlexibleStaging` currently throws an exception when an empty docker directory is provided. It should instead just log a warning.

Also add integration tests for this.

**Comment by patflynn** [comment]
do we need any validation or warning?  wouldn't gcloud spit out warning/errors based off of the presence/absence of Dockerfile with respect to runtime: java/custom?

**Comment by meltsufin** [comment]
Cloud SDK does error out if you use `runtime: custom` in `app.yaml` and no Dockerfile.

```
ERROR: (gcloud.app.deploy) You must provide your own Dockerfile when using a custom runtime.  Otherwise provide a "runtime" field with one of the supported runtimes.
```

**Comment by patflynn** [comment]
what if you have runtime: java with a Dockerfile?

**Comment by meltsufin** [comment]
Also an error:
```
ERROR: (gcloud.app.deploy) There is a Dockerfile in the current directory, and the runtime field in /tmp/test123/app.yaml is currently set to [runtime: java]. To use your Dockerfile to build a custom runtime, set the runtime field in /tmp/test123/app.yaml to [runtime: custom]. To continue using the [java] runtime, please omit the Dockerfile from this directory.
```

**Comment by patflynn** [comment]
So can we just remove all validation from core? It sounds like a warning
won't do much good as they'll just get a failure from gcloud later.

**Comment by meltsufin** [comment]
The warning would only be useful for someone running stage by itself, without deploy.

**Comment by patflynn** [comment]
@loosebazooka 
@meltsufin 

In any case for runtime: java we shouldn't be staging the docker directory.  I guess that should be a Gradle/Maven bug right?

**Comment by loosebazooka** [comment]
fixed by #324
